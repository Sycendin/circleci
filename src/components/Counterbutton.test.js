import renderer from 'react-test-renderer'
import CounterButton from './CounterButton'
import React from 'react'

it('expect to render counterbutton', () =>{
    const mockColor = 'red';
    const button = renderer.create(<CounterButton color={mockColor}/>).toJSON()
    expect(button).toMatchSnapshot();
})


it('correctly increments counter', () => {
    const mockColor = 'red';
    const component = renderer.create(<CounterButton color={mockColor}/>)
    const instance = component.root;
    const button = instance.findByType("button");
    button.props.onClick()
    button.props.onClick()
    expect(button.props.children[1]).toBe(2)
    expect(button.props.color).toBe('red') 

})


it('button color is correct', () => {
    const mockColor = 'red';
    const component = renderer.create(<CounterButton color={mockColor}/>)
    const instance = component.root;
    const button = instance.findByType("button");
    expect(button.props.color).toBe('red')  

})