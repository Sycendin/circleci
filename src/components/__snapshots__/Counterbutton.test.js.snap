// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render counterbutton 1`] = `
<button
  color="red"
  onClick={[Function]}
>
  count: 
  0
</button>
`;
